app/src/components/pages/ScheduleGenerate.tsx

```diff
@@ -34,13 +34,21 @@ import { LoadingContext } from 'contexts/LoadingContext';
 import { ErrorContext } from 'contexts/ErrorContext';
 import { ISections } from 'interfaces/ScheduleGenerate.interfaces';
 import { CourseInput, Term } from 'types/api.types';
+import { useLocation } from 'react-router-dom';
+
+interface IStateProps {
+  semesterPassed: Term;
+  yearPassed: Date;
+}
 
 function ScheduleGenerate() {
   const loadingContext = useContext(LoadingContext);
   const errorContext = useContext(ErrorContext);
 
-  const [term, setTerm] = useState<string>(Term.All);
-  const [year, setYear] = useState<number>(2022);
+  const { state } = useLocation();

```

Comment:
Nice work, but you have not accounted for the case where a user navigates from any other page to the Schedule Generate page - would set initial values to null or potentially crash the app. Have you tested this use case?